### PR TITLE
Working dir issue.

### PR DESCRIPTION
**Pull request by lyonlh**
1. Fix regex misatch of head and tail whitespace.

2. Fix working dir stack bug.

### CODE REVIEW

File: compiledb-gen-parser [hunk 1]
```diff
@@ -70,8 +70,8 @@ def parse_flags(build_log, proj_dir, inc_prefix, verbose):
     cpp_compile_regex = re.compile("(.*-?[gc]\+\+ )|(.*-?clang\+\+ )")
 
     # Leverage make --print-directory option
-    make_enter_dir=re.compile("make\[\d+\]: Entering directory [`\'\"](?P<dir>.*)[`\'\"]$")
-    make_leave_dir=re.compile("make\[\d+\]: Leaving directory .*$")
+    make_enter_dir=re.compile("[\t ]*make\[\d+\]: Entering directory [`\'\"](?P<dir>.*)[`\'\"][\t ]*$")
+    make_leave_dir=re.compile("[\t ]*make\[\d+\]: Leaving directory .*$")
```
Comment (nickdiego) [inline compiledb-gen-parser]: Can we benmore flexible here? Use `^\s*make...` intead.

Comment (lyonlh) [reply compiledb-gen-parser]: Yes, \s is the better one.

**Comment by lyonlh** [comment]
Please don't merge the request, it has a debug log to be removed.

**Comment by nickdiego** [comment]
Oops, too late..
Please, send out a new PR removing it.

**Pull request merged**